flake: test-e2e updateWorkspace.spec.ts - update workspace with ephemeral parameter enabled

Seen here: https://github.com/coder/coder/actions/runs/16596131640/job/46943111766#step:11:1752

May be related to https://github.com/coder/coder/commit/1320b8d5be0cabb2493608fa5cf5c3016d23db68

```
==> Errors
/home/runner/work/coder/coder/site/e2e/helpers.ts:79:
   at helpers.ts:79

  77 | 	await page.getByLabel("Password").fill(options.password);
  78 | 	await page.getByRole("button", { name: "Sign In" }).click();
> 79 | 	await expectUrl(page).toHavePathName("/workspaces");
     | 	                      ^
  80 | 	// biome-ignore lint/suspicious/noExplicitAny: update once logged in
  81 | 	(ctx as any)[Symbol.for("currentUser")] = options;
  82 | }
Error: The page does not have the expected URL pathname.
Expected: "/workspaces"
Actual: "/login"
==> Attachments
{
  name: 'video',
  path: '/home/runner/work/coder/coder/site/test-results/tests-workspaces-updateWor-798d8-ephemeral-parameter-enabled-tests/video.webm',
  contentType: 'video/webm',
  body: undefined
}
==> Tests failed
==> Tests that require a license were skipped, because no license was provided
35 passed
58 skipped
1 failed
  /home/runner/work/coder/coder/site/e2e/tests/workspaces/updateWorkspace.spec.ts › update workspace with ephemeral parameter enabled
 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

--- a/site/e2e/helpers.ts
+++ b/site/e2e/helpers.ts
@@ -903,23 +903,28 @@ const fillParameters = async (
 		}
 
 		// Use modern locator approach instead of waitForSelector
-		const parameterLabel = page.getByTestId(`parameter-field-${richParameter.name}`);
+		const parameterLabel = page.getByTestId(
+			`parameter-field-${richParameter.name}`,
+		);
 		await expect(parameterLabel).toBeVisible();
 
 		if (richParameter.type === "bool") {
-			const parameterField = parameterLabel.getByTestId('parameter-field-bool')
+			const parameterField = parameterLabel
+				.getByTestId("parameter-field-bool")
 				.locator(`.MuiRadio-root input[value='${buildParameter.value}']`);
 			await parameterField.click();
 		} else if (richParameter.options.length > 0) {
-			const parameterField = parameterLabel.getByTestId('parameter-field-options')
+			const parameterField = parameterLabel
+				.getByTestId("parameter-field-options")
 				.locator(`.MuiRadio-root input[value='${buildParameter.value}']`);
 			await parameterField.click();
 		} else if (richParameter.type === "list(string)") {
 			throw new Error("not implemented yet"); // FIXME
 		} else {
 			// text or number
-			const parameterField = parameterLabel.getByTestId('parameter-field-text')
-				.locator('input');
+			const parameterField = parameterLabel
+				.getByTestId("parameter-field-text")
+				.locator("input");
 			await parameterField.fill(buildParameter.value);
 		}
 	}
@@ -1238,14 +1243,12 @@ export const disableDynamicParameters = async (
 
 	// Wait for the success message or page to update
 	await page
-	.locator(
-		"[role='alert']:has-text('Template updated successfully')",
-	)
-	.first()
-	.waitFor({
-		state: "visible",
-		timeout: 15000,
-	});
+		.locator("[role='alert']:has-text('Template updated successfully')")
+		.first()
+		.waitFor({
+			state: "visible",
+			timeout: 15000,
+		});
 
 	// Additional wait to ensure the changes are persisted
 	await page.waitForTimeout(500);

--- a/site/e2e/helpers.ts
+++ b/site/e2e/helpers.ts
@@ -127,6 +127,10 @@ export const createWorkspace = async (
 	const name = randomName();
 	await page.getByLabel("name").fill(name);
 
+	if (buildParameters.length > 0) {
+		await page.waitForSelector("form", { state: "visible" });
+	}
+
 	await fillParameters(page, richParameters, buildParameters);
 
 	if (useExternalAuth) {
@@ -898,28 +902,24 @@ const fillParameters = async (
 			);
 		}
 
-		const parameterLabel = await page.waitForSelector(
-			`[data-testid='parameter-field-${richParameter.name}']`,
-			{ state: "visible" },
-		);
+		// Use modern locator approach instead of waitForSelector
+		const parameterLabel = page.getByTestId(`parameter-field-${richParameter.name}`);
+		await expect(parameterLabel).toBeVisible();
 
 		if (richParameter.type === "bool") {
-			const parameterField = await parameterLabel.waitForSelector(
-				`[data-testid='parameter-field-bool'] .MuiRadio-root input[value='${buildParameter.value}']`,
-			);
+			const parameterField = parameterLabel.getByTestId('parameter-field-bool')
+				.locator(`.MuiRadio-root input[value='${buildParameter.value}']`);
 			await parameterField.click();
 		} else if (richParameter.options.length > 0) {
-			const parameterField = await parameterLabel.waitForSelector(
-				`[data-testid='parameter-field-options'] .MuiRadio-root input[value='${buildParameter.value}']`,
-			);
+			const parameterField = parameterLabel.getByTestId('parameter-field-options')
+				.locator(`.MuiRadio-root input[value='${buildParameter.value}']`);
 			await parameterField.click();
 		} else if (richParameter.type === "list(string)") {
 			throw new Error("not implemented yet"); // FIXME
 		} else {
 			// text or number
-			const parameterField = await parameterLabel.waitForSelector(
-				"[data-testid='parameter-field-text'] input",
-			);
+			const parameterField = parameterLabel.getByTestId('parameter-field-text')
+				.locator('input');
 			await parameterField.fill(buildParameter.value);
 		}
 	}
@@ -1217,10 +1217,14 @@ export const disableDynamicParameters = async (
 		waitUntil: "domcontentloaded",
 	});
 
+	await page.waitForSelector("form", { state: "visible" });
+
 	// Find and uncheck the "Enable dynamic parameters" checkbox
 	const dynamicParamsCheckbox = page.getByRole("checkbox", {
 		name: /Enable dynamic parameters for workspace creation/,
 	});
+
+	await dynamicParamsCheckbox.waitFor({ state: "visible" });
 
 	// If the checkbox is checked, uncheck it
 	if (await dynamicParamsCheckbox.isChecked()) {
@@ -1228,11 +1232,21 @@ export const disableDynamicParameters = async (
 	}
 
 	// Save the changes
-	await page.getByRole("button", { name: /save/i }).click();
+	const saveButton = page.getByRole("button", { name: /save/i });
+	await saveButton.waitFor({ state: "visible" });
+	await saveButton.click();
 
 	// Wait for the success message or page to update
-	await page.waitForSelector("text=Template updated successfully", {
+	await page
+	.locator(
+		"[role='alert']:has-text('Template updated successfully')",
+	)
+	.first()
+	.waitFor({
 		state: "visible",
-		timeout: 10000,
+		timeout: 15000,
 	});
+
+	// Additional wait to ensure the changes are persisted
+	await page.waitForTimeout(500);
 };

